frontend/src/utils/utils.ts

```diff
@@ -488,6 +488,31 @@ export const prettyBytesOrNA = function (n: number) {
     return prettyBytes(n);
 }
 
+
+/**
+ * Determines if two sets are equal.
+ *
+ * This function checks if two sets (xs and ys) have the same size and
+ * the same elements. It assumes that the sets contain elements of type T.
+ * Equality is determined by checking if every element in set xs is also
+ * present in set ys.
+ *
+ * @template T - The type of elements in the sets.
+ * @param {Set<T>} xs - The first set to be compared.
+ * @param {Set<T>} ys - The second set to be compared.
+ * @returns {boolean} - Returns `true` if the sets are equal, otherwise returns `false`.
+ * @example
+ * // returns true
+ * eqSet(new Set([1, 2, 3]), new Set([3, 2, 1]));
+ *
+ * @example
+ * // returns false
+ * eqSet(new Set([1, 2, 3]), new Set([4, 5, 6]));
+ */
+export const eqSet = <T = string,>(xs: Set<T>, ys: Set<T>): boolean =>
```

Comment:
This is a classic mistake, this fails for 

[1, 1, 2]  and [1, 2, 3]

Since size is just array size and not "number of unique elements", they have the same size and checking only the first one one by one, and checking if the second one contains all the first elements, they would be considered equal.

Proper set equality is a bit more difficult and (cpu)time consuming

Reply:
@rikimaru0345 but this function is for `Set` not for arrays. Typescript is guarding that.

Reply:
The problem would be if somebody silence Typescript and provides an array instead. We could be defensive and use `new Set([...xs])` in the implementation to handle both Set and Arrays and then compare sizes.

But that only happens if somebody silences TS and uses `eqSet` for arrays instead of Set.

Reply:
I see, should be good then! :)